src/utils/workspace.ts

```diff
@@ -78,3 +82,21 @@ export function getContainingWorkspace(fsPath: string): vscode.WorkspaceFolder |
         return fsUtils.isPathEqual(f.uri.fsPath, fsPath) || fsUtils.isSubpath(f.uri.fsPath, fsPath);
     });
 }
+
+export async function getDeploymentWorkspace(target?: vscode.Uri | string | SlotTreeItemBase): Promise<string> {
+
+    if (target instanceof vscode.Uri) {
```

Comment:
Per comments in https://github.com/microsoft/vscode-azuretools/pull/573, you moved this logic directly into the shared package. Still waiting for you to update this PR based on that before I can approve